src/main/java/net/minecraftforge/common/extensions/IForgeDispensibleContainerItem.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.DispensibleContainerItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.BlockHitResult;
+import org.jetbrains.annotations.Nullable;
+
+public interface IForgeDispensibleContainerItem
+{
+    private DispensibleContainerItem self()
+    {
+        return (DispensibleContainerItem)this;
+    }
+
+    /**
+     * ItemStack sensitive version of {@link DispensibleContainerItem#emptyContents(Player, Level, BlockPos, BlockHitResult)}
+     * Tries to place the content of the container into the level as a block.
```

Comment:
That's just the current usecase by vanilla. As evident by the name of the method, all it does it attempt to empty the contents of the container and returns whether it was successful. The position of the block is provided; however, that could still mean the contents were emptied into another block.

Reply:
done